const httpError = require('../../helpers/httpError');
const { Remainder } = require('../../models/remainderModel');

async function deleteRemaindersController(req, res, next) {
  const { id } = req.params;
  try {
    const remainder = await Remainder.findByIdAndRemove(id);
    if (!remainder) {
      return next(httpError(404, 'No Remainder found'));
    }
    return res.status(200).json(remainder);
  } catch (error) {
    return res.status(500).json({ error: 'Internal server error' });
  }
}

module.exports = { deleteRemaindersController };
